plugins/events-backend/src/service/hub/createEventBusRouter.ts

```diff
@@ -0,0 +1,278 @@
+  return new MemoryEventBusStore();
+}
+
+/**
+ * Creates a new event bus router
+ * @internal
+ */
+export async function createEventBusRouter(options: {
+  logger: LoggerService;
+  database: DatabaseService;
+  scheduler: SchedulerService;
+  lifecycle: LifecycleService;
+  httpAuth: HttpAuthService;
+  notifyTimeoutMs?: number; // for testing
+}): Promise<Handler> {
+  const { httpAuth, notifyTimeoutMs = DEFAULT_NOTIFY_TIMEOUT_MS } = options;
+  const logger = options.logger.child({ type: 'EventBus' });
+  const router = Router();
+
+  const store = await createEventBusStore(options);
+
+  const apiRouter = await createOpenApiRouter();
+
+  router.use(apiRouter);
+
+  apiRouter.post('/bus/v1/events', async (req, res) => {
+    const credentials = await httpAuth.credentials(req, {
+      allow: ['service'],
+    });
+    const topic = req.body.event.topic;
+    const notifiedSubscribers = req.body.notifiedSubscribers;
+    const result = await store.publish({
+      event: {
+        topic,
+        eventPayload: req.body.event.payload,
+      } as EventParams,
+      notifiedSubscribers: req.body.notifiedSubscribers,
+    });
+    if (result) {
+      logger.info(`Published event to '${topic}' with ID '${result.eventId}'`, {
+        subject: credentials.principal.subject,
+      });
+      res.status(201).end();
+    } else {
+      if (notifiedSubscribers) {
+        const notified = `'${notifiedSubscribers.join("', '")}'`;
+        logger.info(
+          `Skipped publishing of event to '${topic}', subscribers have already been notified: ${notified}`,
+          { subject: credentials.principal.subject },
+        );
+      } else {
+        logger.info(
+          `Skipped publishing of event to '${topic}', no subscribers present`,
+          { subject: credentials.principal.subject },
+        );
+      }
+      res.status(204).end();
+    }
+  });
+
+  apiRouter.get(
+    '/bus/v1/subscriptions/:subscriptionId/events',
+    async (req, res) => {
+      const credentials = await httpAuth.credentials(req, {
+        allow: ['service'],
+      });
+      const id = req.params.subscriptionId;
+
+      const controller = new AbortController();
+      req.on('end', () => controller.abort());
+
+      // By setting up the listener first we make sure we don't miss any events
+      // that are published while reading. If an event is published we'll receive
+      // a notification, which we may ignore depending on the outcome of the read
+      const listener = await store.setupListener(id, {
+        signal: controller.signal,
+      });
+
+      // By timing out requests we make sure they don't stall or that events get stuck.
+      // For the caller there's no difference between a timeout and a
+      // notification, either way they should try reading again.
+      const timeout = setTimeout(() => {
+        controller.abort();
+      }, notifyTimeoutMs);
+
+      try {
+        const { events } = await store.readSubscription(id);
```

Comment:
Current max batch size is 10, with the processing all events before requesting new ones. Yeah if the client is particularly slow at processing events I think it should persist the result itself rather than doing any parallelization here.